### DISCUSSION TITLE
Allow server actions on export mode

### DISCUSSION THREAD

**Discussion by artrixdotdev** [vercel/next.js#67336] [Ideas]
### Goals

1.  Allow complete separation from the backend and frontend in the same codebase
2. Having a less half baked experience when using Next with [Tauri](https://tauri.app/) and [Electron](https://www.electronjs.org/)
3. Enable server-side functionality in static exports without requiring the server to be in the same place

### Non-Goals

1. Replace or modify existing server actions
2. Change the core architecture of Next.js
3. Provide full server capabilities in a client-only environment

### Background

Server actions in Next.js have become a cornerstone feature, but their absence in export mode creates a significant limitation for developers building desktop applications with Tauri or Electron, or creating static sites that need dynamic functionality. Currently, the primary alternative is to bypass server actions entirely and make direct API calls to your Next.js app via things like `fetch('https://myapp.vercel.app')`  . This workaround, while working completely fine, introduces unnecessary complexity and deviates from the streamlined Next.js development experience. Implementing server actions support in export mode would bridge the gap between static exports and dynamic functionality, allowing developers to leverage Next.js's powerful features while maintaining the benefits of static generation. This could significantly expand Next.js's utility across a broader range of applications, from JAMstack sites to desktop apps, aligning with the framework's goal of providing a flexible, powerful solution for diverse use cases.

### Proposal

Adding a `baseURL` to `next.config.js` as an optional property could be a simple solution. Here's how it might work:

When someone tries to use server-side features in a static export, they'd need to add a baseURL in the config or set it as an environment variable (like NEXT_BASE_URL). Instead of calling `POST /signin`, it would use `POST ${baseURL}/signin`. This would be super helpful for folks using desktop frameworks with Next.js to create a single codebase that works as both an app and a website.

For other use cases, there might be CORS issues, but those can be easily fixed by tweaking the `headers()` property in the config. This approach gives developers the flexibility to use server actions in static exports without complicating the Next.js setup too much.

This can also apply to anything else that requires a direct interaction with the server and client, including `next/image`, `next/link`, and whatever else might need it.

I admit I don't know the Next.js codebase very well but if someone can point me to the correct files I would love to make a PR and get this working.

**Comment by Fustigate8933** [comment]
yes. it is hell turning my nextjs app that uses server side features like cookies into electron desktop app. please add this feature.
